src/decorator/relations/ManyToOne.ts

```diff
@@ -55,5 +55,24 @@ export function ManyToOne<T>(typeFunctionOrTarget: string|((type?: any) => Objec
             inverseSideProperty: inverseSideProperty,
             options: options
         } as RelationMetadataArgs);
+
+        if (options && options.index) {
+            const indexOptions: IndexOptions & { name?: string } = typeof options.index === 'object' ? options.index : {};
+
+            getMetadataArgsStorage().indices.push({
+                target: object.constructor,
+                name: indexOptions.name ? indexOptions.name : undefined,
+                columns: [propertyName],
+                synchronize: indexOptions && (indexOptions as { synchronize: false }).synchronize === false ? false : true,
+                where: indexOptions ? indexOptions.where : undefined,
+                unique: indexOptions && indexOptions.unique ? true : false,
+                spatial: indexOptions && indexOptions.spatial ? true : false,
+                fulltext: indexOptions && indexOptions.fulltext ? true : false,
+                parser: indexOptions ? indexOptions.parser : undefined,
+                sparse: indexOptions && indexOptions.sparse ? true : false,
+                background: indexOptions && indexOptions.background ? true : false,
+                expireAfterSeconds: indexOptions ? indexOptions.expireAfterSeconds : undefined
+            } as IndexMetadataArgs);
```

Comment:
I suggest you reuse the existing `Index()` decorator here so we don't duplicate.
```suggestion
            Index(indexOptions)(object, propertyName);
```

Reply:
Done. Also added logic for handling custom index names without code duplication.